File: 5.GPT/JS/1.intro.js
require('dotenv').config({path: '../.env'}) // .env 파일 읽어서 메모리에 올리기
const axios = require('axios');

const openaiApiKey = process.env.OPENAI_API_KEY;

// const url = 'https://api.openai.com/v1/responses';
const url = 'https://api.openai.com/v1/chat/completions';

async function getGPTResponse() {
  const response = await axios.post(url, {
    // 본문
    "model": "gpt-3.5-turbo",
    "messages": [
      // {'role': 'system', 'content': 'you are a helpful assistant.'},
      {'role': 'system', 'content': 'you are a cook.'},
      // {'role': 'system', 'content': 'you are a software engineer.'},
      // {'role': 'system', 'content': 'you are a singer.'},
      {'role': 'user', 'content': '나 배고파 대한민국에서 봄 제철 음식 추천해줘.'}
    ],
    temperature: 1.0,  // 정확도 (창의성)
    top_p: 0.9,  // 확률 기반 토큰 선택 범위
    frequency_penalty: 0.5, // 반복 억제
    presence_penalty: 0.6, // 얼마나 새로운 주제를 가져올거냐... 등등
    max_tokens: 100  // 답변의 길이 결정
  },
  { // 헤더
    headers: {
      "Content-Type": "application/json",
      "Authorization": `Bearer ${openaiApiKey}`
    }
    })
  
  // console.log(response.data);
  return response.data.choices[0].message;
}

async function myfunction() {
  const ai_response = await getGPTResponse();
  console.log(ai_response);
}

myfunction();

File: 5.GPT/PYTHON/12.claude/1.intro.py
# pip install anthropic

import os
import anthropic
from dotenv import load_dotenv

load_dotenv()
api_key = os.getenv("ANTHROPIC_API_KEY")

# client = openai 했던 것처럼...
client = anthropic.Anthropic(api_key=api_key)

message = client.messages.create(
  model="claude-3-7-sonnet-20250219",
  max_tokens=1000,
  temperature=0.7,
  messages=[
    # {"role": "user", "content": "안년하세요"},
    # {"role": "user", "content": "python으로 flask 만들고 싶어. 가장 간단한 예제 보여줘."},
    # {"role": "user", "content": "GPT란 무엇인지 간단하게 주요 기능 알려줘"},
    {"role": "user", "content": "개발자 주인공 웹툰 스토리 만들어줘."},
  ]
)

print(message.content[0].text)